src/main/drivers/serial_uart_impl.h

```diff
@@ -144,13 +161,21 @@ typedef struct uartHardware_s {
 #if !defined(STM32F7)
     uint8_t af;
 #endif
-#if defined(STM32F7)
+#if defined(STM32F7) || defined(STM32H7)
+    uint8_t txIrq;
     uint8_t rxIrq;
 #else
     uint8_t irqn;
 #endif
     uint8_t txPriority;
     uint8_t rxPriority;
+
+#ifdef STM32H7
+    volatile uint8_t *txBuffer;
+    volatile uint8_t *rxBuffer;
+    uint16_t txBufferSize;
```

Comment:
is size really necessary? It's hardcoded for other CPUs

Reply:
IIRC, there was a feature proposal to do run-time allocation of buffer spaces... Let me see ... yeah, #3650. I think the sizes are here for in preparation for the feature.

Reply:
Isn't uart_hardware in FLASH? `uartDevice is place to store it ..

Reply:
Oh wait. It was a quick solution not to expose the size constant in `serialUART` (`serial.c`); H7 code uses buffers DMA_RAM region in D2_SRAM where it is MPU controlled for DMA r/w.
Need some review later.